modules/nw-ingress-controller-nodeportservice-projects.adoc

```diff
@@ -0,0 +1,190 @@
+----
+
+. Use the `oc new-project` command to create a new project:
++
+[source,terminal]
+----
+$ oc new-project <project_name>
+----
+
+. Enter the following command to label the new namespace:
++
+[source,terminal]
+----
+$ oc label namespace <project_name> <key>=<value> <1>
+----
+<1> Where `<key>=<value>` must match the value that you specified in the `namespaceSelector` section of your Ingress Controller CR.
+
+. Create a new application in your cluster:
++
+[source,terminal]
+----
+$ oc new-app --image=<image_name> <1>
+----
+<1> An example of `<image_name>` is `quay.io/openshifttest/hello-openshift:multiarch`.
+
+. Create a `Route` object for a service, so that the pod can use the service to expose the application external to the cluster.
++
+[source,terminal]
+----
+$ oc expose svc/<service_name> --hostname=<svc_name>.<project_name>.<domain_name> <1>
+----
++
+[NOTE]
+====
+You must specify the domain name of your custom Ingress Controller in the `--hostname` argument. If you do not do this, the Ingress Operator uses the default Ingress Controller to serve all the routes for your cluster. 
+====
+
+. Check that the route has the `Admitted` status and that it includes metadata for the custom Ingress Controller:
++
+[source,terminal]
+----
+$ oc get route/hello-openshift -o json | jq '.status.ingress
+----
++
+.Example output
+[source,terminal]
+----
+# ...
+{
+  "conditions": [
+    {
+      "lastTransitionTime": "2024-05-17T18:25:41Z",
+      "status": "True",
+      "type": "Admitted"
+    }
+  ],
+  [
+    {
+      "host": "hello-openshift.nodeportsvc.ipi-cluster.example.com",
+      "routerCanonicalHostname": "router-nodeportsvc.nodeportsvc.ipi-cluster.example.com",
+      "routerName": "nodeportsvc", "wildcardPolicy": "None"
+    }
+  ],
+}
+----
+
+. Update the default `IngressController` CR to prevent the default Ingress Controller from managing the `NodePort`-type `Service`. The default Ingress Controller will continue to monitor all other cluster traffic.
++
+[source,terminal]
+----
+$ oc patch --type=merge -n openshift-ingress-operator ingresscontroller/default --patch '{"spec":{"namespaceSelector":{"matchExpressions":[{"key":"<key>","operator":"NotIn","values":["<value>]}]}}}'
+----
+
+.Verification
+
+. Verify that the wildcard DNS record is managed by the Ingress Controller by entering the following command in your CLI.
++
+[source,terminal]
+----
+$ oc get route
+----
++
+.Output example
+[source,terminal]
+----
+NAME              HOST/PORT                                             PATH   SERVICES          PORT       TERMINATION   WILDCARD 
+hello-openshift   hello-openshift.nodeportsvc.ipi-cluster.example.com          hello-openshift   8080-tcp   
+----
+
+. Check the connection of your cluster to verify that your cluster uses the IP addresses from external DNS servers for DNS resolution:
++
+[source,terminal]
+----
+$ oc curl hello-openshift.nodeportsvc.ipi-cluster.example.com:<port> 
```

Comment:
1. It is `curl http://hello-openshift.nodeportsvc.ipi-cluster.example.com:<port>` not `oc curl ...` 
2.  should provide description of  `<port>`
```
Port is the nodeport we get from the nodeport service, `80:32432/TCP` means we should use `32432` here.
``` 
See Line 83 and 88, `80:32432/TCP,443:31366/TCP` means for http route we should use nodeport 32432, for https should use nodeport 31366

Reply:
:heavy_check_mark: